app/models/user.rb

```diff
@@ -293,10 +293,34 @@ def teacher?
     given_lectures.any?
   end
 
+  def teachable_editor?
+    edited_courses.any? || edited_lectures.any?
+  end
+
+  def teachable_editor_or_teacher?
+    teachable_editor? || teacher?
+  end
+
+  def can_edit_teachables?
+    admin? || teachable_editor_or_teacher?
+  end
+
+  # if you are not a teacher of lecture or a module editor,
+  # but just an additional editor of some lecture, you
+  # will not be considered active if all of your edited lectures
+  # are older than one year
+  def active_teachable_editor?
+    return false unless can_edit_teachables?
+    return true if admin || course_editor? || teacher?
+
+    edited_lectures.select { |l| l.term.nil? || !stale? }
```

Comment:
Why do we select lectures that don't have a `nil` term here as well? What does that mean for a lecture, why is term optional? Is it omitted for lectures that are term independent (`lecture.course.term_independent`)? What does that mean semantically?

Reply:
There are lectures that do not belong to a term, like the "Helpdesk" (this has several advantages: you do not need to resubscribe as a student in a new term, materials for these should be independent of the term anyway and some others). 
By the curent code, they are considered stale. This is wrong, I think. I think the code in `older_than?` should return false then; we could then remove the `l.term.nil?` condition here (and I would have to look if the logic still fits everywhere).

Reply:
Ok, but then I think the way it is right now is fine. Otherwise, you include logic in `older_than?` that is not even concerned with specifying is something is older than something else violating the single-responsibility principle.